pkgs/applications/networking/cluster/kcp/default.nix

```diff
@@ -0,0 +1,72 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  testers,
+# kubernetes-kcp,
+}:
+
+let
+  rev = "6a4e71a9d35c6906d3fdd46595e8b693c38669f5";
+in
+buildGoModule rec {
+  pname = "kubernetes-kcp";
+  version = "0.26.0";
+
+  src = fetchFromGitHub {
+    owner = "kcp-dev";
+    repo = "kcp";
+    rev = "v${version}";
+    sha256 = "sha256-ZEgDeILo2weSAZgBsfR2EQyzym/I/+3P99b47E5Tfrw=";
+  };
+  vendorHash = "sha256-IONbTih48LKAiEPFNFdBkJDMI2sjHWxiqVbEJCskyio=";
+
+  subPackages = [ "cmd/kcp" ];
+
+  # TODO: The upstream has the additional version information pulled from go.mod
+  # dependencies.
+  ldflags = [
+    "-X k8s.io/client-go/pkg/version.gitCommit=${rev}"
+    "-X k8s.io/client-go/pkg/version.gitTreeState=clean"
+    "-X k8s.io/client-go/pkg/version.gitVersion=v${version}"
+    # "-X k8s.io/client-go/pkg/version.gitMajor=${KUBE_MAJOR_VERSION}"
+    # "-X k8s.io/client-go/pkg/version.gitMinor=${KUBE_MINOR_VERSION}"
+    "-X k8s.io/client-go/pkg/version.buildDate=unknown"
+    "-X k8s.io/component-base/version.gitCommit=${rev}"
+    "-X k8s.io/component-base/version.gitTreeState=clean"
+    "-X k8s.io/component-base/version.gitVersion=v${version}"
+    # "-X k8s.io/component-base/version.gitMajor=${KUBE_MAJOR_VERSION}"
+    # "-X k8s.io/component-base/version.gitMinor=${KUBE_MINOR_VERSION}"
+    "-X k8s.io/component-base/version.buildDate=unknown"
+  ];
+
+  # TODO: Check if this is necessary.
+  # __darwinAllowLocalNetworking = true;
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    $out/bin/kcp completion bash > kcp.bash
+    $out/bin/kcp completion zsh > kcp.zsh
+    $out/bin/kcp completion fish > kcp.fish
+    installShellCompletion kcp.{bash,zsh,fish}
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    # package = kubernetes-kcp;
+    command = "kcp --version";
+    # NOTE: Once the go.mod version is pulled in, the version info here needs
+    # to be also updated.
+    version = "v${version}";
+  };
+
+  meta = with lib; {
```

Comment:
`with lib` is deprecated #208242

Reply:
Ah thanks for the catch, I checked out other Kubernetes related package and got it over here. This is fixed in the latest commit (squashed / force-pushed).